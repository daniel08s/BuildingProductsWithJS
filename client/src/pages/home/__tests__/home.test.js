/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import HomePage, {Home} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '1',
  login: 'testuser',
  registrationDate: new Date(2018, 1, 1, 1, 1, 1, 1),
};

test('# HomePage', () => {
  const store = mockStore({auth: {user}, questions: {questions: []}});

  const wrapper = shallow(<HomePage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Home', () => {
  const questions = [];
  const fetchQuestions = () => {
    questions.push({
      id: 1,
      owner: user,
      text: 'Test question',
      answers: [{answer: 'Test answer'}],
    });
  };
  const doAnswer = () => {};
  const component = (
    <Home
      user={user}
      questions={questions}
      fetchQuestions={fetchQuestions}
      doAnswer={doAnswer}
    />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
