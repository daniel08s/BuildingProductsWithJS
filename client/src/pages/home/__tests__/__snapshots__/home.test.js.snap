// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Home 1`] = `
<div>
  <Component
    current="/"
    user={
      Object {
        "id": "1",
        "login": "testuser",
        "registrationDate": 2018-02-01T01:01:01.001Z,
      }
    }
  />
  <div
    className="container"
  >
    <Connect(Question)
      onAnswer={[Function]}
      question={
        Object {
          "answers": Array [
            Object {
              "answer": "Test answer",
            },
          ],
          "id": 1,
          "owner": Object {
            "id": "1",
            "login": "testuser",
            "registrationDate": 2018-02-01T01:01:01.001Z,
          },
          "text": "Test question",
        }
      }
    />
  </div>
</div>
`;

exports[`# HomePage 1`] = `
<Home
  doAnswer={[Function]}
  fetchQuestions={[Function]}
  questions={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={
    Object {
      "id": "1",
      "login": "testuser",
      "registrationDate": 2018-02-01T01:01:01.001Z,
    }
  }
/>
`;
